### DISCUSSION TITLE
Does portable-pty always utf-8?

### DISCUSSION THREAD

**Discussion by jjangga0214** [wezterm/wezterm#2463] [Q&A]
Hi!

Does `portable-pty` always read in utf-8?

Thank you in advance.

**Comment by wez** [comment]
the `Read` and `Write` impls are byte streams. There is no portably defined encoding at the pty layer.
The interpretation of the byte stream encoding is application-defined, rather than defined by portable-pty.

The application spawned into the pty will typically use system-provided functions to determine the encoding of its input/output; on Windows, it will generally use the system code page settings, while on unix systems, the locale is usually selected based on the environment variables passed to the application.
